# Relampago population genomics
Code and supplementary information used for genome assembly of Parvodontia relampaga and analysis of its population structure, ecology, and evolution. Each folder represents one major step of the analysis. In each folder you will find:
  1) A markdown that summarises the commands and programs used, my thought process behind using them, and occasional trouble shooting information
  2) The full scripts used for each analysis/process


note to self. 
one paper describing population strucutre, naivte or not? host-pathogen genotyper interactions  
one paper desribing migration patterns, genes involved in local adaptation, and ecological niche modeling   
# need to make picture of flow chart bruh 
# Refer to and cite this in approriate folders https://speciationgenomics.github.io/

Note - With the availability of several hundreds to tens of thousands of loci in the population genomic realm, neutral loci can be differentiated from adaptive loci by calculating locus-statistics such as the fixation index (FST), which is a statistical measure of population differentiation based upon genetic distance. Loci with outlier FST values are likely to be under selection (adaptive loci) and hence should be removed from the analyses for making accurate inferences about demographic processes (Gr√ºnwald et al. 2016). ## Need to double check that the SNP filtering does this prior to PCA etc etc. 
